e2e: add a test case of ROX PVC-PVC clone

Madhu-1 opened the pull request:
adding a test case to test the below case of RBD

* Create PVC and application
* Create PVC-PVC ROX clone
* Try to create many pods using the ROX PVC
* Try to write the data and verify only read access.

Madhu-1 committed:
e2e: add a test case of ROX PVC-PVC clone

adding a test case to test below case for RBD

* Create PVC and application
* Scale down application pod
* Create PVC-PVC ROX clone
* Try to create many pods that use the ROX PVC
* Try to write the data and verify only read
access.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

iPraveenParihar commented on e2e/rbd.go:
```diff
@@ -3375,6 +3375,118 @@ var _ = Describe("RBD", func() {
+				smartClonePVC.Spec.DataSource.Name = pvc.Name
+				smartClonePVC.Spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.ReadOnlyMany}
+				smartClonePVC.Spec.DataSource.Name = pvc.Name
+				err = createPVCAndvalidatePV(f.ClientSet, smartClonePVC, deployTimeout)
```
not needed, assigned already in line 3412

Madhu-1 commented:
https://jenkins-ceph-csi.apps.ocp.cloud.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.31/detail/mini-e2e-helm_k8s-1.31/140/pipeline/100 E2E passed here

iPraveenParihar commented on e2e/rbd.go:
```diff
@@ -3375,6 +3375,117 @@ var _ = Describe("RBD", func() {
+						appClone.Namespace,
+						&opt)
+					readOnlyErr := fmt.Sprintf("cannot create %s: Read-only file system", filePath)
+					if !strings.Contains(stdErr, readOnlyErr) {
```
@Madhu-1,  I think proper error checking is required here. 
From logs, It seems like it failed to execute command. 

Am not sure, if the error is because of Read-only?

```
  I1028 11:03:05.307462 84561 exec_util.go:83] ExecWithOptions: execute(POST https://192.168.49.2:8443/api/v1/namespaces/rbd-341/pods/rbd-3410/exec?command=%2Fbin%2Fsh&command=-c&command=echo+%27Hello+World%27+%3E+%2Fvar%2Flib%2Fwww%2Fhtml%2Ftest&container=web-server&container=web-server&stderr=true&stdout=true)
  I1028 11:03:05.740366 84561 pod.go:238] failed to execute command: command terminated with exit code 2
  I1028 11:03:05.740409 84561 pod.go:327] command echo 'Hello World' > /var/lib/www/html/test failed: failed to execute command: command terminated with exit code 2
  I1028 11:03:05.743216 84561 exec_util.go:59] ExecWithOptions {Command:[/bin/sh -c echo 'Hello World' > /var/lib/www/html/test] Namespace:rbd-341 PodName:rbd-3411 ContainerName:web-server Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
  I1028 11:03:05.743243 84561 util.go:499] >>> kubeConfig: /root/.kube/config
  I1028 11:03:05.744249 84561 exec_util.go:66] ExecWithOptions: Clientset creation
  I1028 11:03:05.744307 84561 exec_util.go:83] ExecWithOptions: execute(POST https://192.168.49.2:8443/api/v1/namespaces/rbd-341/pods/rbd-3411/exec?command=%2Fbin%2Fsh&command=-c&command=echo+%27Hello+World%27+%3E+%2Fvar%2Flib%2Fwww%2Fhtml%2Ftest&container=web-server&container=web-server&stderr=true&stdout=true)
  I1028 11:03:05.808732 84561 pod.go:238] failed to execute command: command terminated with exit code 2
  I1028 11:03:05.808777 84561 pod.go:327] command echo 'Hello World' > /var/lib/www/html/test failed: failed to execute command: command terminated with exit code 2
  I1028 11:03:05.843965 84561 exec_util.go:59] ExecWithOptions {Command:[/bin/sh -c echo 'Hello World' > /var/lib/www/html/test] Namespace:rbd-341 PodName:rbd-3412 ContainerName:web-server Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
  I1028 11:03:05.844003 84561 util.go:499] >>> kubeConfig: /root/.kube/config
  I1028 11:03:05.844976 84561 exec_util.go:66] ExecWithOptions: Clientset creation
  I1028 11:03:05.845031 84561 exec_util.go:83] ExecWithOptions: execute(POST https://192.168.49.2:8443/api/v1/namespaces/rbd-341/pods/rbd-3412/exec?command=%2Fbin%2Fsh&command=-c&command=echo+%27Hello+World%27+%3E+%2Fvar%2Flib%2Fwww%2Fhtml%2Ftest&container=web-server&container=web-server&stderr=true&stdout=true)
  I1028 11:03:06.144433 84561 pod.go:238] failed to execute command: command terminated with exit code 2
  I1028 11:03:06.144474 84561 pod.go:327] command echo 'Hello World' > /var/lib/www/html/test failed: failed to execute command: command terminated with exit code 2
  I1028 11:03:06.150257 84561 pod.go:416] Waiting for pod rbd-3410 to be deleted
  I1028 11:03:08.159622 84561 pod.go:416] Waiting for pod rbd-3411 to be deleted
  I1028 11:03:10.167370 84561 pod.go:416] Waiting for pod rbd-3412 to be deleted
```

Madhu-1 replied:
we are checking the stdErr, if there any message other than the required message the test fails. the above logs are logging only the error not the stdErr.

Madhu-1 replied:
added one more commit to log the stderror for debugging we don't need to check anything in error check here.

iPraveenParihar replied:
Thanks for addressing that.

Madhu-1 committed:
e2e: log stderror in execCommandInPodAndAllowFail

If the command we try to execute fails, log
the stdError along with error for better debugging.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

iPraveenParihar reviewed (approved):
LGTM

nixpanic commented:
@mergifyio queue
